modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaFieldWidgetTest.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content_featured_media_field\FunctionalJavascript;
+
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\file\Entity\File;
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\media\Entity\Media;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+use Drupal\Tests\node\Traits\ContentTypeCreationTrait;
+use Drupal\Tests\node\Traits\NodeCreationTrait;
+
+/**
+ * Tests the output of "oe_featured_media_widget" widget.
+ *
+ * @group oe_content_featured_media_field
+ */
+class FeaturedMediaFieldWidgetTest extends WebDriverTestBase {
+
+  use ContentTypeCreationTrait;
+  use NodeCreationTrait;
+  use MediaTypeCreationTrait;
+
+  /**
+   * A field storage to use in this test class.
+   *
+   * @var \Drupal\field\Entity\FieldStorageConfig
+   */
+  protected $fieldStorage;
+
+  /**
+   * The field used in this test class.
+   *
+   * @var \Drupal\field\Entity\FieldConfig
+   */
+  protected $field;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'field',
+    'field_ui',
+    'media',
+    'media_test_source',
+    'oe_media',
+    'oe_content_featured_media_field',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->createContentType(['type' => 'page']);
+
+    $user = $this->drupalCreateUser([
+      'access content',
+      'create page content',
+    ]);
+    $this->drupalLogin($user);
+
+    // Create an image file.
+    \Drupal::service('file_system')->copy($this->root . '/core/misc/druplicon.png', 'public://example.jpg');
+    $image = File::create(['uri' => 'public://example.jpg']);
+    $image->save();
+
+    // Create a media entity of image media type.
+    $media_entity = Media::create([
+      'bundle' => 'image',
+      'name' => 'Test image',
+      'field_media_image' => [
+        [
+          'target_id' => $image->id(),
+          'alt' => 'default alt',
+          'title' => 'default title',
+        ],
+      ],
+    ]);
+    $media_entity->save();
+
+    // Create another media entity.
+    $media_entity = Media::create([
+      'bundle' => 'document',
+      'name' => 'Test media',
+    ]);
+    $media_entity->save();
+
+    $this->fieldStorage = FieldStorageConfig::create([
```

Comment:
I don't like too much the idea of using properties to store field instances and such, I think they belong to the test method itself most of the time. But talking about this case, it's clear we don't need to store them in properties. `$this->field` is not even used, while `$this->fieldStorage` is used once and to get its name (which we define once) so we can refer to it directly by its name `featured_media_field`.